jumpscale/sals/nettools/__init__.py

```diff
@@ -32,117 +47,167 @@ def tcp_connection_test(ipaddr: str, port: int, timeout: Optional[int]):
     """
     conn = None
     try:
-        conn = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        if timeout:
-            conn.settimeout(timeout)
-        try:
-            conn.connect((ipaddr, port))
-        except BaseException:
-            return False
-    finally:
-        if conn:
-            conn.close()
-    return True
+        conn = socket.create_connection((ipaddr, port), timeout)
+    except (socket.gaierror, socket.herror):
+        # raised for address-related errors
+        raise
```

Comment:
can we raise the exception here ? or log the message to make debugging easier or we return False ?

Reply:
'can we raise the exception here ?'
the exception would raise already.

'log the message to make debugging easier or we return False'
logging sounds perfect, but not sure if we should return false address-related errors. more discussion?

Reply:
I mean raise with proper message like

```
raise j.exceptions.Runtime("An error <descritption> occured with the socket, <something to do>")
```